lib/topic_query.rb

```diff
@@ -971,8 +984,18 @@ def self.remove_muted_tags(list, user, opts = {})
           SELECT 1
             FROM topic_tags tt
            WHERE tt.tag_id IN (:tag_ids)
-             AND tt.topic_id = topics.id)",
+             AND tt.topic_id = topics.id
+             #{user && !opts[:skip_categories] ? "AND COALESCE(category_users.notification_level, :regular) < :watching_or_infinite" : ""})",
           tag_ids: muted_tag_ids,
+          regular: CategoryUser.notification_levels[:regular],
+          watching_or_infinite:
+            (
+              if SiteSetting.watched_precedence_over_muted
```

Comment:
I feel like this is at odds with the `remove_muted_tags_from_latest` site setting? I don't feel like we need to consider `watched_precedence_over_muted` site setting because `remove_muted_tags_from_latest` already states that muted tags are removed regardless of category watch precedence.